### ISSUE TITLE
[release/2.12] Segmentation fault running speech_transformer torchbench model with amp_bf16 on XPU

### ISSUE THREAD

**Issue by mengfei25** [intel/torch-xpu-ops#3434]
### 🐛 Describe the bug

When running the `speech_transformer` torchbench model with `inductor` backend on XPU using **PyTorch 2.12 RC wheel** (`amp_bf16`, inference), the process crashes with a **Segmentation fault** during warmup.

The crash occurs after Dynamo emits a graph break warning due to `Tensor.item()` in the model code. No Python traceback is produced — the process is killed by SIGSEGV.

#### Error Log

```
loading model: 0it [00:02, ?it/s]
xpu  eval  speech_transformer
W0423 03:56:24.105000 161925 torch/_dynamo/variables/tensor.py:1612] [0/0] Graph break from `Tensor.item()`, consider setting:
W0423 03:56:24.105000 161925 torch/_dynamo/variables/tensor.py:1612] [0/0]     torch._dynamo.config.capture_scalar_outputs = True
W0423 03:56:24.105000 161925 torch/_dynamo/variables/tensor.py:1612] [0/0] or:
W0423 03:56:24.105000 161925 torch/_dynamo/variables/tensor.py:1612] [0/0]     env TORCHDYNAMO_CAPTURE_SCALAR_OUTPUTS=1
W0423 03:56:24.105000 161925 torch/_dynamo/variables/tensor.py:1612] [0/0] to include these operations in the captured graph.
W0423 03:56:24.105000 161925 torch/_dynamo/variables/tensor.py:1612] [0/0]
W0423 03:56:24.105000 161925 torch/_dynamo/variables/tensor.py:1612] [0/0] Graph break: from user code at:
W0423 03:56:24.105000 161925 torch/_dynamo/variables/tensor.py:1612] [0/0]   File "/workspace/pytorch-src/benchmarks/dynamo/torchbench.py", line 477, in forward_pass
W0423 03:56:24.105000 161925 torch/_dynamo/variables/tensor.py:1612] [0/0]     return mod(*inputs)
W0423 03:56:24.105000 161925 torch/_dynamo/variables/tensor.py:1612] [0/0]   File "/workspace/benchmark-src/torchbenchmark/models/speech_transformer/speech_transformer/transformer/transformer.py", line 27, in forward
W0423 03:56:24.105000 161925 torch/_dynamo/variables/tensor.py:1612] [0/0]     encoder_padded_outputs, *_ = self.encoder(padded_input, input_lengths)
W0423 03:56:24.105000 161925 torch/_dynamo/variables/tensor.py:1612] [0/0]   File "/workspace/benchmark-src/torchbenchmark/models/speech_transformer/speech_transformer/transformer/encoder.py", line 60, in forward
W0423 03:56:24.105000 161925 torch/_dynamo/variables/tensor.py:1612] [0/0]     non_pad_mask = get_non_pad_mask(padded_input, input_lengths=input_lengths)
W0423 03:56:24.105000 161925 torch/_dynamo/variables/tensor.py:1612] [0/0]   File "/workspace/benchmark-src/torchbenchmark/models/speech_transformer/speech_transformer/utils/utils.py", line 108, in get_non_pad_mask
W0423 03:56:24.105000 161925 torch/_dynamo/variables/tensor.py:1612] [0/0]     non_pad_mask[i, input_lengths[i] :] = 0
W0423 03:56:24.105000 161925 torch/_dynamo/variables/tensor.py:1612] [0/0]
W0423 03:56:24.105000 161925 torch/_dynamo/variables/tensor.py:1612] [0/0]
Segmentation fault
```

#### Analysis

- The graph break occurs at `get_non_pad_mask()` where `input_lengths[i]` triggers `Tensor.item()`, causing Dynamo to split the graph.
- After the graph break, subsequent compilation or execution of the partial graph on XPU leads to a SIGSEGV.
- A segfault (no Python traceback) suggests the crash is in a native/C++ layer — possibly in the XPU runtime, oneDNN, or the Triton-compiled kernel execution.

#### Reproducer

```bash
docker run -it -e TZ=Aisa/Shanghai --device=/dev/mem --device=/dev/dri --group-add video --privileged --shm-size=8g intelgpu/ubuntu-24.04-lts2:2523.40 bash

curl -LsSf https://astral.sh/uv/install.sh | env UV_INSTALL_DIR="/usr/local/bin" sh
uv venv myvenv --python 3.12 --clear
source myvenv/bin/activate
uv pip install pip numpy 'setuptools<81' wheel

uv pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/test/xpu  # 2.12

pytorch_commit="$(python -c 'import torch; print(torch.version.git_version)')"
git clone https://github.com/pytorch/pytorch
cd pytorch
git checkout ${pytorch_commit}
rm -rf torch torch-xpu-ops
git clone https://github.com/intel/torch-xpu-ops
rsync -avz ./torch-xpu-ops/.ci/benchmarks/ ./benchmarks/dynamo/

uv pip install pip pandas psutil scipy requests

git clone https://github.com/pytorch/benchmark tmp-benchmark
cd tmp-benchmark
uv pip install -r requirements.txt
python install.py speech_transformer

python ./benchmarks/dynamo/torchbench.py --performance --amp --amp-dtype bfloat16 --inference -d xpu -n 10 --only speech_transformer --backend=inductor --cold-start-latency --timeout 10800 --disable-cudagraphs
```

### Versions

PyTorch 2.12 RC XPU wheel (from `https://download.pytorch.org/whl/test/xpu`)
Torch commit: `489deee763382be6f9ccb296797b782d7054a06a`
Triton: 3.7.1
Python 3.12
Docker image: `intelgpu/ubuntu-24.04-lts2:2523.40`

**Comment by etaf** [comment]
What‘s the last good commit?

**Comment by mengfei25** [comment]
related issue https://github.com/intel/torch-xpu-ops/issues/2057

**Comment by etaf** [comment]
Guilty commit： https://github.com/pytorch/pytorch/pull/177158

**Comment by etaf** [comment]
Fix PR: https://github.com/pytorch/pytorch/pull/181753

**Comment by etaf** [comment]
The similar issue also found in cuda at https://github.com/pytorch/pytorch/issues/180741, and the guilty commit has been reverted by https://github.com/pytorch/pytorch/pull/181873, and cherrying-pick to 2.12 release branch.
